gradle build issue

## Steps to Reproduce

1. Run these commands against the android emulator:

```
flutter create repro
cd repro
flutter run
```

=>

```
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/mit/tmp/repro5/android/app/build.gradle' line: 11

* What went wrong:
A problem occurred evaluating project ':app'.
> Flutter SDK not found. Define location with flutter.sdk in the local.properties file.

```

I think I am seeing the same thing on Windows with the gallery:

```shell
C:\src\flutter\examples\flutter_gallery [master]> flutter run
Running 'flutter packages get' in flutter_gallery...  7.1s
FAILURE: Build failed with an exception.

* Where:
Build file 'C:\src\flutter\examples\flutter_gallery\android\app\build.gradle' line: 11

* What went wrong:
A problem occurred evaluating project ':app'.
> Flutter SDK not found. Define location with flutter.sdk in the local.properties file.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
Error running gradle: Exit code 1 from: C:\Program Files\Android\Android Studio\gradle\gradle-2.14.1\bin\gradle.bat app:properties
FAILURE: Build failed with an exception.

* Where:
Build file 'C:\src\flutter\examples\flutter_gallery\android\app\build.gradle' line: 11

* What went wrong:
A problem occurred evaluating project ':app'.
> Flutter SDK not found. Define location with flutter.sdk in the local.properties file.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
Error running gradle: Exit code 1 from: C:\Program Files\Android\Android Studio\gradle\gradle-2.14.1\bin\gradle.bat app:properties
Launching lib/main.dart on Nexus 6P in debug mode...
Running 'gradle assembleDebug'...                    31.3s
Built android\app\build/outputs/apk/app-debug.apk (31.2MB).
Uninstalling old version...
Installing android\app\build/outputs/apk/app.apk...   6.8s
Syncing files to device...                            3.0s

🔥  To hot reload your app on the fly, press "r" or F5. To restart the app entirely, press "R".
The Observatory debugger and profiler is available at: http://127.0.0.1:8100/
For a more detailed help message, press "h" or F1. To quit, press "q", F10, or Ctrl-C.
```

Interestingly, everything still seems to work as expected despite the scary error message.

--- a/packages/flutter_tools/lib/src/android/gradle.dart
+++ b/packages/flutter_tools/lib/src/android/gradle.dart
@@ -69,6 +69,7 @@ String get gradleAppOut {
 
 String get gradleAppOutDirV2 {
   final String gradle = ensureGradle();
+  ensureLocalProperties();
   try {
     final String properties = runCheckedSync(
       <String>[gradle, 'app:properties'],
@@ -128,8 +129,8 @@ String ensureGradle() {
   return gradle;
 }
 
-Future<Null> buildGradleProject(BuildMode buildMode, String target) async {
-  // Create android/local.properties.
+/// Create android/local.properties if needed.
+File ensureLocalProperties() {
   final File localProperties = fs.file('android/local.properties');
   if (!localProperties.existsSync()) {
     localProperties.writeAsStringSync(
@@ -137,6 +138,11 @@ Future<Null> buildGradleProject(BuildMode buildMode, String target) async {
         'flutter.sdk=${_escapePath(Cache.flutterRoot)}\n'
     );
   }
+  return localProperties;
+}
+
+Future<Null> buildGradleProject(BuildMode buildMode, String target) async {
+  final File localProperties = ensureLocalProperties();
   // Update the local.properties file with the build mode.
   // FlutterPlugin v1 reads local.properties to determine build mode. Plugin v2
   // uses the standard Android way to determine what to build, but we still

